lib/miq_expression/target.rb

```diff
@@ -10,10 +10,10 @@ def self.parse!(field)
   # {:model_name => Host<ApplicationRecord> , :associations => ['vms']<Array>, :column_name => 'host_name' <String>}
   def self.parse_params(field)
     return unless field.kind_of?(String)
-    match = self::REGEX.match(field) || return
-    # convert matches to hash to format
-    # {:model_name => 'User', :associations => ...}
-    parsed_params = Hash[match.names.map(&:to_sym).zip(match.to_a[1..-1])]
```

Comment:
This craziness is so much more simply written as , so I've changed that wherever needed.

```diff
- parsed_params = Hash[match.names.map(&:to_sym).zip(match.to_a[1..-1])]
+ parsed_params = match.named_captures.symbolize_keys
```